lua/nvim-tree/watcher.lua

```diff
@@ -74,4 +74,11 @@ end
 
 M.Watcher = Watcher
 
+function M.purge_watchers()
+  for _, watcher in pairs(M._watchers) do
+    watcher:stop()
+  end
+  M._watchers = {}
```

Comment:
Pushed this up.

We were keeping the old watchers and so the new ones would not start.